### PR TITLE
feat:kakarot connectors

### PR DESCRIPTION
**Pull request by enitrat**
Initial work on implementing a wrapper around EVM connectors through Kakarot

### CODE REVIEW

File: packages/kakarot/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# or
+yarn add @starknet-react/kakarot
+```
+
+## Usage
+
+Simply import the kakarotConnectors.
+
+```ts
+import { kakarotConnectors } from '@starknet-react/kakarot`
```
Comment (fracek) [inline packages/kakarot/README.md:18]:
```suggestion
import { kakarotConnectors } from "@starknet-react/kakarot"
```

**Review by fracek — APPROVED** [review]

**Review by fracek — CHANGES_REQUESTED** [review]
Looks good. Can you fix the CI and the README so that I can merge and make a release?

File: packages/kakarot/src/kakarot.ts [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+let chainChanged: EthereumConnectorEvents["onChainChanged"] | undefined;
+let connect: EthereumConnectorEvents["onConnect"] | undefined;
+let disconnect: EthereumConnectorEvents["onDisconnect"] | undefined;
+
+export interface KakarotConnectorOptions {
+  id: string;
+  name: string;
+  icon: ConnectorIcons;
+}
+export class KakarotConnector extends Connector {
```
Comment (Eikix) [inline packages/kakarot/src/kakarot.ts:82]: as I understand, some methods comply with starknet wallet spec, via starknet react, other methods comply with Eth EIP1193

Should we add in docstring methods linked with Starknet and others for Eth?

File: packages/core/src/connectors/base.ts [hunk 3]
```diff
@@ -45,7 +45,9 @@ export abstract class Connector extends EventEmitter<ConnectorEvents> {
   /** Whether connector is already authorized */
   abstract ready(): Promise<boolean>;
   /** Connect wallet. */
-  abstract connect(): Promise<ConnectorData>;
+  abstract connect(parameters?: {
+    chainIdHint: bigint;
+  }): Promise<ConnectorData>;
```
Comment (fracek) [inline packages/core/src/connectors/base.ts:50]:
```suggestion
  abstract connect(parameters: {
    chainIdHint?: bigint;
  } = {}): Promise<ConnectorData>;
```

**Review by fracek — APPROVED** [review]

**Pull request merged**